### PR TITLE
docs: add "permissions.content: write" to GH workflow examples

### LINKED ISSUE #8876: GitHub actions requires more permissions
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#issues).

### Description

Since https://github.blog/changelog/2023-02-02-github-actions-updating-the-default-github_token-permissions-to-read-only/ it is required to configure the repo (or the workflow) to have permissions to publish GH pages. See https://github.com/gohugoio/hugoDocs/pull/1991 for example.

### Self-service

- [X] I'd be willing to address this documentation request myself.

### PR DESCRIPTION
**Pull request by e-minguez**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #0000) and the maintainers have approved on my working plan.

## Motivation

Fixes #8876

## Test Plan

None

### Test links

N/A

## Related issues/PRs

#8876

### CODE REVIEW

**Review by Josh-Cena — CHANGES_REQUESTED** [review]
Thank you! I don't think the issue link is necessary (you can either remove it or replace it with something like `# Explicitly grant write permission`), but otherwise this should be good.

File: website/docs/deployment.mdx [hunk 1]
```diff
@@ -390,6 +390,10 @@ on:
     # Review gh actions docs if you want to further define triggers, paths, etc
     # https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#on
 
+# https://github.com/peaceiris/actions-gh-pages/issues/852
```
Comment (slorber) [inline website/docs/deployment.mdx:393]:
```suggestion
```

File: website/docs/deployment.mdx [hunk 2]
```diff
@@ -488,6 +492,10 @@ on:
   push:
     branches: [main]
 
+# https://github.com/peaceiris/actions-gh-pages/issues/852
```
Comment (slorber) [inline website/docs/deployment.mdx:495]:
```suggestion
```

**Review by slorber — COMMENTED** [review]
Thanks 👍 

Agree that the issue link is not needed

**Pull request merged**